pkg/apis/multiclusterdns/v1alpha1/servicednsrecord_types.go

```diff
@@ -25,15 +25,19 @@ import (
 type ServiceDNSRecordSpec struct {
 	// FederationName is the name of the federation to which the corresponding federated service belongs
 	FederationName string `json:"federationName,omitempty"`
```

Comment:
Have you thought about embedding a `FederationDomain` struct type within this spec? However, that could potentially be wasteful when multiple services share the same `FederationName`.

What are your thoughts on making this a non-optional `ObjectReference` such as `domainRef`? That could help alleviate some of the concerns around making sure `FederationDomain` exists first.

Reply:
> Have you thought about embedding a FederationDomain struct type within this spec? However, that could potentially be wasteful when multiple services share the same FederationName.

Ack

> What are your thoughts on making this a non-optional ObjectReference such as domainRef? That could help alleviate some of the concerns around making sure FederationDomain exists first.

That's a good idea. I would update the API, Thanks @font

Reply:
@font, on a second thought i stumbled upon this https://github.com/kubernetes-incubator/service-catalog/issues/1411#issuecomment-337224947, where api-machinery folks suggest to avoid ObjectReference. Also the UX would not be great if we ask user to fillup the federationDomain details, like the UID, kind, etc..

FederationName should be a non-optional field. i would update it.